ldap/servers/plugins/automember/automember.c

```diff
@@ -2434,19 +2510,19 @@ automember_rebuild_task_thread(void *arg)
                         automember_update_membership(config, entries[i], NULL) == SLAPI_PLUGIN_FAILURE)
                     {
                         result = SLAPI_PLUGIN_FAILURE;
-                        automember_config_unlock();
                         goto out;
                     }
                 }
                 list = PR_NEXT_LINK(list);
             }
         }
     }
-    automember_config_unlock();
 
 out:
+    automember_config_unlock();
+
     if (plugin_is_betxn && fixup_pb) {
-        if (i == 0 || result != 0) { /* no updates performed */
+        if ((i == 0 || result != 0) && aborted == 0) { /* no updates performed */
```

Comment:
Should abort txn if i==0 (in other words it should be: if (i==0 || (result !=0 && aborted==0))

Reply:
You're right, but @tbordaz mentioned that the rebuild task actually starts over from the beginning with each rebuild, so there is no real benefit to "keeping" the transaction.  The next rebuild will be just as expensive.  He also said we could change that transaction model to use smaller transactions.  Not sure if that's going to impact performance but I'm going to try it out next...